presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/ExpressionConverter.java

```diff
@@ -175,6 +179,19 @@ private static Object getValue(Type type, Marker marker, ConnectorSession sessio
             return ((Slice) marker.getValue()).getBytes();
         }
 
+        if (type instanceof DecimalType) {
+            DecimalType decimalType = (DecimalType) type;
+            Object value = marker.getValue();
+            if (Decimals.isShortDecimalType(decimalType)) {
+                checkArgument(value instanceof Long, "A short decimal should be represented by a long value but was " + value);
+                return value;
+            }
+            else {
+                checkArgument(value instanceof Slice, "A long decimal should be represented by a Slice value but was " + value);
```

Comment:
`checkArgument` takes string format so use `%s` instead of concatenation. Also, we should print the class since it might not be obvious from the `toString()`
```java
checkArgument(value instanceof Slice, "A long decimal should be represented by a Slice value but was %s", value.getClass().getName());

Reply:
Fixed.